### PR TITLE
MBS-8337: Show kanbans in block activity_modules

### PR DESCRIPTION
**Pull request by fdagner**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -28,6 +28,52 @@
...
+
+echo $OUTPUT->header();
+
+$modulenameplural = get_string('modulenameplural', 'mod_kanban');
+echo $OUTPUT->heading($modulenameplural);
+
+$kanbans = get_all_instances_in_course('kanban', $course);
+
+if (empty($kanbans)) {
+     notice(get_string('no$kanbaninstances', 'mod_kanban'), new moodle_url('/course/view.php', ['id' => $course->id]));
```
Comment (PhMemmel) [inline index.php:49]: The $ in 'no$kanbaninstances' is probably a typo

File: index.php [hunk 2]
```diff
@@ -28,6 +28,52 @@
 
 $id = required_param('id', PARAM_INT);
 
-require_login();
+$course = $DB->get_record('course', ['id' => $id], '*', MUST_EXIST);
+require_course_login($course);
```
Comment (PhMemmel) [inline index.php:32]: Who should be able to see this? Only teacher? There's a capability check missing for this site then.

File: index.php [hunk 3]
```diff
@@ -28,6 +28,52 @@
...
+    $table->head = [get_string('sectionname', 'format_' . $course->format), get_string('name')];
+    $table->align = ['left', 'left'];
+} else {
+    $table->head  = [get_string('name')];
+    $table->align = ['left'];
+}
+
+foreach ($kanbans as $kanban) {
+    $linkcss = null;
+    if (!$kanban->visible) {
```
Comment (PhMemmel) [inline index.php:66]: Not only do you have to check the visibility, but also that the current user has the capability to view the kanban.

Hidden by restrictions for example has to be reflected here, or just in general the capability to view the kanban course module

File: index.php [hunk 4]
```diff
@@ -28,6 +28,52 @@
...
+echo $OUTPUT->heading($modulenameplural);
+
+$kanbans = get_all_instances_in_course('kanban', $course);
+
+if (empty($kanbans)) {
+     notice(get_string('no$kanbaninstances', 'mod_kanban'), new moodle_url('/course/view.php', ['id' => $course->id]));
+}
+
+$usesections = course_format_uses_sections($course->format);
+$table = new html_table();
```
Comment (PhMemmel) [inline index.php:53]: Using a template for displaying the list would be more clean I feel like

**Review by PhMemmel — CHANGES_REQUESTED** [review]

File: index.php (hunk 4 above)
Comment (fdagner) [reply index.php:53]: I try it

Comment (PhMemmel) [reply index.php:53]: other plugins use html table, so just stick to it

File: index.php (hunk 2 above)
Comment (fdagner) [reply index.php:32]: The content is for all users of the course, then require_course_login($course); should be sufficient?

File: index.php [hunk 5]
```diff
@@ -28,6 +28,63 @@
...
+} else {
+    $table->head  = [get_string('name')];
+    $table->align = ['left'];
+}
+
+$kanbanfound = false;
+
+foreach ($kanbans as $kanban) {
+    $context = context_module::instance($kanban->coursemodule, IGNORE_MISSING);
+    if (!$context || !$kanban->visible || !has_capability('mod/kanban:view', $context)) {
```
Comment (PhMemmel) [inline index.php:64]: at this point you will never show invisible kanbans, not even to the teacher which seems to be wrong. The code in l. 71/72 to show the entry "dimmed" will never be triggered.

**Review by PhMemmel — CHANGES_REQUESTED** [review]
one minor issue, besides that looks good

File: lang/en/kanban.php [hunk 6]
```diff
@@ -123,6 +123,14 @@
 $string['messageprovider:due'] = 'Card due';
 $string['messageprovider:moved'] = 'Card moved';
 $string['modulename'] = 'Kanban board';
+$string['modulename_help'] = 'This activity supports using kanban method for managing projects or learning processes.
```
Comment (PhMemmel) [inline lang/en/kanban.php:126]: This is not properly being formatted in the info box, looks like the markdown list is not supported. Could this be solved differently?

Comment (fdagner) [reply lang/en/kanban.php:126]: This actually belongs to #62. Seems to have somehow got into the commit here?

Comment (PhMemmel) [reply lang/en/kanban.php:126]: I was already wondering :)
Can you fix the history please? And fix formatting in #62 :)

File: lang/en/kanban.php [hunk 7]
```diff
@@ -133,6 +133,7 @@
 $string['newcard'] = 'New card';
 $string['newcolumn'] = 'New column';
 $string['nogroupavailable'] = 'No group available';
+$string['nokanbaninstances'] = 'There are no kanban boards in this course';
```
Comment (PhMemmel) [inline lang/en/kanban.php:136]:
```suggestion
$string['nokanbaninstances'] = 'There are no kanban boards in this course or you are not allowed to access them';
```

**Review by PhMemmel — APPROVED** [review]
LGTM, good job!

**Pull request merged**